include/deal.II/base/mpi.templates.h

```diff
@@ -413,6 +414,78 @@ namespace Utilities
       internal::all_reduce(MPI_MIN, values, mpi_communicator, minima);
     }
 
+
+
+    template <typename T>
+    std::vector<T>
+    compute_union(const std::vector<T> &vec, const MPI_Comm &comm)
+    {
+      // 1) collect vector entries and create union
+      std::set<T> result;
+
+      std::vector<T> buffer;
+
+      if (this_mpi_process(comm) == 0)
+        {
+          for (const auto i : vec)
+            result.insert(i);
+
+
+          for (unsigned int i = 1; i < n_mpi_processes(comm); i++)
+            {
+              MPI_Status status;
+              const auto ierr_1 = MPI_Probe(MPI_ANY_SOURCE,
+                                            internal::Tags::compute_union,
+                                            comm,
+                                            &status);
+              AssertThrowMPI(ierr_1);
+
+              int        amount;
+              const auto ierr_2 =
+                MPI_Get_count(&status,
+                              internal::mpi_type_id(vec.data()),
+                              &amount);
+              AssertThrowMPI(ierr_2);
+
+              buffer.resize(amount);
+
+              const auto ierr_3 = MPI_Recv(buffer.data(),
+                                           amount,
+                                           MPI_INT,
+                                           status.MPI_SOURCE,
+                                           status.MPI_TAG,
+                                           comm,
+                                           MPI_STATUS_IGNORE);
+              AssertThrowMPI(ierr_3);
+
+              for (const auto i : buffer)
+                result.insert(i);
```

Comment:
More a remark than a comment: I am pretty sure that accumulating the result in the vector directly would be faster than the set (unless compilers changed the implementation of `std::set`), which is why we use `vector` in the `DynamicSparsityPattern` and other places where performance matters. Here the `set` is the least amount of code, and this operation is so much cheaper than the MPI communication, so this is definitely the correct solution.